usr/src/cmd/text-install/osol_install/text_install/zpool_selection.py

```diff
@@ -0,0 +1,312 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2009, 2010, Oracle and/or its affiliates. All rights reserved
+# Copyright 2017 Alexander Pyhalov
+#
+
+'''
+Screens and functions to display a list of pools to the user.
+'''
+
+import curses
+import logging
+import re
+
+from osol_install.text_install import _, RELEASE
+from osol_install.text_install.base_screen import BaseScreen, \
+                                                  SkipException, \
+                                                  UIMessage
+from osol_install.text_install.disk_window import get_minimum_size, \
+                                                  get_recommended_size
+from osol_install.text_install.edit_field import EditField
+from osol_install.text_install.error_window import ErrorWindow
+from osol_install.text_install.i18n import fit_text_truncate, \
+                                           textwidth, \
+                                           ljust_columns
+from osol_install.text_install.list_item import ListItem
+from osol_install.text_install.multi_list_item import MultiListItem
+from osol_install.text_install.scroll_window import ScrollWindow
+from osol_install.text_install.window_area import WindowArea
+from osol_install.text_install.ti_install_utils import get_zpool_list, \
+                                                       get_zpool_free_size, \
+                                                       get_zpool_be_names
+class ZpoolScreen(BaseScreen):
+    '''
+    Allow the user to select a (valid) zpool target for installation
+    '''
+    
+    HEADER_TEXT = _("Pools")
+    PARAGRAPH = _("Where should %(release)s be installed?") % RELEASE
+    SIZE_TEXT = _("Recommended size:  %(recommend).1fGB      "
+                  "Minimum size: %(min).1fGB")
+    POOL_SEEK_TEXT = _("Seeking pools on system")
+    TOO_SMALL = _("Too small")
+    NO_POOLS = _("No pools found. ")
```

Comment:
really, just copied https://github.com/OpenIndiana/slim_source/blob/oi/hipster/usr/src/cmd/text-install/osol_install/text_install/disk_selection.py#L76 and removed irrelevant part.